Automattic/woocommerce-payments
# Fix transactions search view when it has too many search terms

Fixes #8781 

The PR adds a CSS workaround  to fix the UX of the `transactions search` component, when there is a large number of search terms. ( Case-in-point - The landing link of `Total payment volume` on the Payment activity widget ) .  The PR makes the search bar clearly visible whenever the tag list is focused. 

#### Notes
This is a temporary fix for the issue via a CSS tweak. This has been done to cater to the immediate requirements of the Payment activity widget project, where the issue of having too many search terms becomes more obvious. Else, it would be somewhat uncommon for most merchants at this stage. 

A more permanent solution would require fixing the `TableCard` component at WooCommerce core ( https://github.com/woocommerce/woocommerce/issues/32239 ) , or will need changes at WooPayments components level ( https://github.com/Automattic/woocommerce-payments/issues/2269 ).

#### Screenshots / Recording



https://github.com/Automattic/woocommerce-payments/assets/4162931/f3ab35f5-09f1-40b0-88f5-fe00ae15da84



#### Testing instructions
* Check out this branch
* Browse to  this link : 
`http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&filter=advanced&store_currency_is=usd&date_between%5B0%5D=2024-06-14&date_between%5B1%5D=2024-06-20&search%5B0%5D=charge&search%5B1%5D=payment&search%5B2%5D=payment_failure_refund&search%5B3%5D=payment_refund&search%5B4%5D=refund&search%5B5%5D=refund_failure&search%5B6%5D=dispute&search%5B7%5D=dispute_reversal&search%5B8%5D=card_reader_fee&search%5B9%5D=adjustment`

* Change the website address in the above link if needed. 
* If it does not work, browse to `Payments > Transactions` and add several terms in the search box until it becomes very long
* When you try to add more terms, the search box should appear below in a separate row. 
* When you try to type new terms, it should show the button below it clearly
* When the search box is active, you should see scroll bars when you click and drag the tags. 
* Try this with a mobile sized screen too. 
-------------------

- [x] Covered with tests (or have a good reason not to test in description ☝️) - no new tests needed

**Post merge**
